### PR TITLE
fix unclear docs useing scatter() instead of circle()

### LINKED ISSUE #14144: [FEATURE] Unclear docs on "first_steps_2.rst"
### Problem description

I was running through the docs and became confused on [this stage](https://github.com/bokeh/bokeh/blob/branch-3.7/docs/bokeh/source/docs/first_steps/first_steps_2.rst#rendering-circles).

You tell us to "switch a line() to a circle()"... however, the code shows no mention of the `circle()` being used.  Instead, you use `scatter()`.  

### Feature description

Clean up this area so it makes a bit more sense.  Either show us how to use `circle()` or show us how to use `scatter()`.

### Potential alternatives

Leave it as is.  Maybe it doesn't need to be changed?

### Additional information

Thanks!

### PR DESCRIPTION
**Pull request by mosc9575**
- [x] issues: fixes #14144

### CODE REVIEW

File: src/bokeh/plotting/glyph_api.py [hunk 1]
```diff
@@ -1115,7 +1115,8 @@ def scatter(self, *args: Any, **kwargs: Any) -> GlyphRenderer:
 
             size (str or list[float]) : values or field names of sizes in |screen units|
 
-            marker (str, or list[str]): values or field names of marker types
+            marker (str, or list[str]): values or field names of marker types, visit
+                :class:`~bokeh.core.enums.MarkerType` for a complete list of valid values.
```
Comment (mattpap) [inline src/bokeh/plotting/glyph_api.py:1119]: Perhaps this could use an update after PR #14165.

Comment (mosc9575) [reply src/bokeh/plotting/glyph_api.py:1119]: Comment to myself: https://github.com/bokeh/bokeh/blob/branch-3.7/docs/bokeh/source/docs/user_guide/basic/scatters.rst needs an update, too.

**Comment by mattpap** [comment]
Looks like one more case of missing `radius` or wrong glyph method being used: https://github.com/bokeh/bokeh/issues/14196.

**Comment by bryevdv** [comment]
Right I don't think that code is actually executed anywhere it's just inline in the docs

**Comment by mosc9575** [comment]
This is ready for review.

I searched the docs for the use of `circle()` and updated the calls with a value for `radius` or replaced it with `scatter`, depending on the documentation above and below. I did so, because of https://github.com/bokeh/bokeh/issues/13899.

I did not touch 

https://github.com/bokeh/bokeh/blob/5be848f0e7fa34301cd7b390426328dbf7a78d6c/docs/bokeh/source/docs/user_guide/server/app.rst?plain=1#L528

to avoid a conflict with [#14197](https://github.com/bokeh/bokeh/pull/14197).

**Review by mattpap — APPROVED** [review]

**Comment by mattpap** [comment]
Test failures are unrelated.

**Comment by mattpap** [comment]
@mosc9575, thanks.

**Pull request merged**